Assignment2_JUNG.Rmd

```diff
@@ -0,0 +1,248 @@
+
+#gather classes columns into rows
+library(tidyr)
+personClass <- gather(personClass, classNumber, classNames, Classes.Q1:Classes.Q5)
+
+#get rid of all the spaces to reduce irrecularities
+personClass[] <- lapply(personClass, function(x) as.character(gsub(" ", "", x)))
+
+#rename the columns
+names(personClass) <- c("studentNames", "classNumbers","classNames")
+
+#remove the classNumber columns and all the NA values
+personClass <- subset(personClass, select = c("studentNames","classNames"))
+personClass <- na.omit(personClass)
+
+#remove all the HUDK4050 (class that all of us are taking) for simplicity
+personClass <- filter(personClass, classNames != "HUDK4050")
+
+#add a column with count
+countPC <- count(personClass, studentNames, classNames)
+names(countPC) <- c("studentNames", "classNames", "count")
+
+#spread 
+countPCdf<- spread(countPC, classNames, count)
+
+#convert it to matrix
+countPCMat <-data.matrix(countPCdf, rownames.force = NA)
+rownames(countPCMat) <- c(countPCdf$studentNames)
+countPCMat[is.na(countPCMat)] <- 0
+personClassMat <- countPCMat[,-c(1) ]
+
+
+##-----Person-Person Matrix-----##
+
+#transpose countPCMat into countCPMat
+classPersonMat <- t(personClassMat)
+
+#create person-person matrix by multiplying person-class and class-person matrices
+personPersonMat <- personClassMat %*% classPersonMat 
+
+#diagnonals of person-person matrix gives no meaningful info (how many classes each person takes - no relation to other students) therefore, remove all the diagnonals 
+diag(personPersonMat)<-NA
+
+##-----betweeness centrality-----##
+
+library(igraph)
+library(dplyr)
+
+#plot network graph
+g3 <- graph.adjacency(personPersonMat,mode="undirected")
+plot.igraph(g3,layout=layout.fruchterman.reingold, vertex.size=10,vertex.label.cex=0.3)
+
+#betweenness centrality
+#vertex betweenness
+btwness <- betweenness(g3)
+btwness <- as.data.frame(btwness)
+vertexBtwness <- tibble::rownames_to_column(btwness, "studentNames")
+
+
+##-----**Who is the most central person in the network?**-----##
+centralPerson <- subset(btwness, btwness==max(btwness))
+#the person with highest betweenness value is Lintong.
+
+
+#edge betweenness
+edgeBtwness <- edge_betweenness(g3, e = E(g3), directed = FALSE, weights = NULL)
+edgeBtwness <- as.data.frame(edgeBtwness)
+
+##-----betweeness degree-----##
+#degree centrality: "counts the number of links held by each node and points at individuals who can quickly connect with the wider network... local measure since it does not take into account the rest of the network and the importance you give to its value depends strongly on the network's size"
+degreeCent <- centr_degree(g3,mode="all")
+
+degreeCentdf <- as.data.frame(degreeCent$res)
+degreeCentdf <- cbind(degreeCentdf, vertexBtwness$studentNames)
+names(degreeCentdf) <- c("degree", "studentNames")
+
+highDegreePerson <- subset(degreeCentdf, degree==max(degree))
+
+```
```

Comment:
Overall, good job! You demonstrated excellent data wrangling skills. Try to interpret the network metrics within the context of the dataset.